tests/integrationtests/tst_perfparser.cpp

```diff
@@ -0,0 +1,332 @@
+/*
+  tst_perfparser.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Nate Rogers <nate.rogers@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QObject>
+#include <QTest>
+#include <QDebug>
+#include <QTextStream>
+#include <QProcess>
+#include <QSignalSpy>
+#include <QTemporaryFile>
+#include <QDateTime>
+
+#include "perfrecord.h"
+#include "perfparser.h"
+#include "util.h"
+#include "data.h"
+#include "unistd.h"
+#include "summarydata.h"
+
+namespace {
+template<typename T>
+bool searchForChildSymbol(const T& root, const QString& searchString)
+{
+    if (root.symbol.symbol == searchString) {
+        return true;
+    } else {
+        for (auto entry : root.children) {
+            if (searchForChildSymbol(entry, searchString)) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+QString getUniqueFilePath()
```

Comment:
this should be removed, and instead use QTemporaryFile at the caller side please